atlasdb-config/src/main/java/com/palantir/atlasdb/factory/ServiceDiscoveringAtlasSupplier.java

```diff
@@ -73,6 +76,24 @@ public KeyValueService getKeyValueService() {
         return keyValueService.get();
     }
 
+    private KeyValueService pollOrCreateRawKeyValueService(AtlasDbFactory atlasFactory) {
+        return atlasDbConfig.pollForDataBaseOnStartup() ? pollForKeyValueService(atlasFactory)
+                : atlasFactory.createRawKeyValueService(keyValueServiceConfig, leaderConfig);
+    }
+
+    private KeyValueService pollForKeyValueService(AtlasDbFactory atlasFactory) {
+        int failureCount = 0;
+        while (true) {
+            try {
+                return atlasFactory.createRawKeyValueService(keyValueServiceConfig, leaderConfig);
+            } catch (Exception ex) {
+                failureCount++;
```

Comment:
Ideally we want to log the exception with a stacktrace unless its message matches some whitelist of known error messages.

Imagine if `createRawKeyValueService` fails for some other random reason (like a bug in the code rather than a connection issue) -- we need a way to troubleshoot this.